modules/oxide-components/src/main/ts/components/menu/Menu.tsx

```diff
@@ -0,0 +1,244 @@
+    </div>
+  );
+});
+
+const ToggleMenuItem = forwardRef<HTMLDivElement, ToggleMenuItemProps>(({ id, enabled = true, onSetup, text, icon, iconResolver, active = false, shortcut, onAction }, ref) => {
+  const [ state, setState ] = useState({
+    enabled,
+    active,
+    focused: false,
+    hovered: false,
+  });
+
+  useEffect(() => {
+    setState((prevState) => ({ ...prevState, enabled, active }));
+  }, [ enabled, active ]);
+
+  const api: ToggleMenuItemInstanceApi = useMemo(() => ({
+    isEnabled: () => state.enabled,
+    setEnabled: (newEnabled: boolean) => {
+      setState((prev) => ({ ...prev, enabled: newEnabled }));
+    },
+    isActive: () => state.active,
+    setActive: (newActive: boolean) => {
+      setState((prev) => ({ ...prev, active: newActive }));
+    }
+  }), [ state ]);
+
+  useEffect(() => {
+    if (onSetup) {
+      const teardown = onSetup(api);
+      return () => teardown(api);
+    }
+    return Fun.noop;
+  }, [ onSetup, api ]);
+
+  return (
+    <div
+      id={id}
+      tabIndex={-1}
+      role='menuitemcheckbox'
+      aria-label={text}
+      aria-haspopup={false}
+      aria-disabled={!state.enabled}
+      aria-selected={state.active}
+      onFocus={() => setState({ ...state, focused: true })}
+      onPointerEnter={() => setState({ ...state, hovered: true })}
+      onPointerLeave={() => setState({ ...state, hovered: false })}
+      onBlur={() => setState({ ...state, focused: false })}
+      onClick={() => onAction(api)}
+      className={Bem.element('tox-collection', 'item', {
+        'enabled': state.active,
+        'active': state.focused || state.hovered,
+        'state-disabled': !state.enabled,
+      })}
+      ref={ref}
+    >
+      <div className={Bem.element('tox-collection', 'item-icon')}>
+        {icon && iconResolver && <Icon icon={icon} resolver={iconResolver} />}
+      </div>
+      <div className={Bem.element('tox-collection', 'item-label')}>{text}</div>
+      {shortcut && <div className={Bem.element('tox-collection', 'item-accessory')}>{shortcut}</div>}
+      <div className={Bem.element('tox-collection', 'item-checkmark')} >
+        {iconResolver && <Icon icon={'checkmark'} resolver={iconResolver} />}
+      </div>
+    </div>
+  );
+});
+
+export const Menu: FC<MenuProps> = ({ items, iconResolver, submenusSide = 'right', autoFocus = true }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const firstItemRef = useRef<HTMLDivElement>(null);
+  KeyboardNavigationHooks.useFlowKeyNavigation({
+    containerRef: ref,
+    selector: '.tox-collection__item:not([aria-disabled="true"])',
+    allowHorizontal: false,
+    cycles: false
+  });
+  useEffect(() => {
+    if (autoFocus && firstItemRef.current) {
+      firstItemRef.current.focus();
+    }
+  }, [ autoFocus ]);
+  const itemsWithId = useMemo(() => Arr.map(items, (itemProps) => ({ ...itemProps, id: Id.generate('menu-item') })), [ items ]);
```

Comment:
There is a built in id generator in react I think we used that for other things.